Client/Extensions/UIImageViewExtensions.swift

```diff
@@ -9,46 +9,55 @@ import Shared
 
 public extension UIImageView {
 
-    public func setIcon(_ icon: Favicon?, forURL url: URL?, completed completionBlock: ((UIColor, URL?) -> Void)? = nil ) {
+    public func setIcon(_ icon: Favicon?, forURL url: URL?, completed completion: ((UIColor, URL?) -> Void)? = nil ) {
+        func finish(filePath: String?, bgColor: UIColor) {
+            if let filePath = filePath {
+                self.image = UIImage(contentsOfFile: filePath)
+            }
+            // If the background color is clear, we may decide to set our own background based on the theme.
+            let color = bgColor.components.alpha < 0.01 ? UIColor.theme.general.faviconBackground : bgColor
+            self.backgroundColor = color
+            completion?(color, url)
+        }
+
         if let url = url, let defaultIcon = FaviconFetcher.getDefaultIconForURL(url: url) {
-            self.image = UIImage(contentsOfFile: defaultIcon.url)
-            self.backgroundColor = defaultIcon.color
-            completionBlock?(defaultIcon.color, url)
+            finish(filePath: defaultIcon.url, bgColor: defaultIcon.color)
         } else {
             let imageURL = URL(string: icon?.url ?? "")
             let defaults = defaultFavicon(url)
             self.sd_setImage(with: imageURL, placeholderImage: defaults.image, options: []) {(img, err, _, _) in
                 guard let image = img, let dUrl = url, err == nil else {
-                    self.backgroundColor = defaults.color
-                    completionBlock?(defaults.color, url)
+                    finish(filePath: nil, bgColor: defaults.color)
                     return
                 }
-                self.color(forImage: image, andURL: dUrl, completed: completionBlock)
+                self.color(forImage: image, andURL: dUrl) { color in
+                    finish(filePath: nil, bgColor: color)
+                }
             }
         }
     }
 
    /*
     * Fetch a background color for a specfic favicon UIImage. It uses the URL to store the UIColor in memory for subsequent requests.
     */
-    private func color(forImage image: UIImage, andURL url: URL, completed completionBlock: ((UIColor, URL?) -> Void)? = nil) {
+    private func color(forImage image: UIImage, andURL url: URL, completed completionBlock: ((UIColor) -> Void)? = nil) {
```

Comment:
This code should only call back with a color, not the URL passed in which is never mutated.